File: devcamp2018/app/src/main/java/com/psato/devcamp/di/component/ViewModelSubComponent.java
package com.psato.devcamp.di.component;

import com.psato.devcamp.presentation.MVVM.QueryViewModelArc;
import com.psato.devcamp.presentation.home.HomeFragmentViewModel;

import dagger.Subcomponent;

@Subcomponent
public interface ViewModelSubComponent {
    @Subcomponent.Builder
    interface Builder {
        ViewModelSubComponent build();
    }

    HomeFragmentViewModel homeFragmentViewModel();
    QueryViewModelArc queryViewModelArc();
}

File: devcamp2018/app/src/main/java/com/psato/devcamp/di/component/ViewModelSubComponent.kt
package com.psato.devcamp.di.component

import com.psato.devcamp.presentation.MVVM.QueryViewModelArc
import com.psato.devcamp.presentation.home.HomeFragmentViewModel

import dagger.Subcomponent

@Subcomponent
interface ViewModelSubComponent {
    @Subcomponent.Builder
    interface Builder {
        fun build(): ViewModelSubComponent
    }

    fun homeFragmentViewModel(): HomeFragmentViewModel
    fun queryViewModelArc(): QueryViewModelArc
}
